<img src="https://raw.githubusercontent.com/rhatiro/Curso_EBAC-Profissao_Cientista_de_Dados/main/ebac-course-utils/media/logo/newebac_logo_black_half.png" alt="ebac-logo">

---

# **SQL para Análise de Dados**
## **Módulo 8** | SQL: Projeto

Professora [Mariane Neiva](https://www.linkedin.com/in/mariane-neiva/)<br>
Aluno [Guilherme Lima](https://www.linkedin.com/in/guilherme-lima-0b7941182/)<br>



---

# **Índice**

- [**0. Conjunto de dados**](#0.-Conjunto-de-dados)  

- [**1. Criação da tabela**](#1.-Criação-da-tabela)  

- [**2. EDA (Exploratory Data Analysis)**](#2.-EDA-(Exploratory-Data-Analysis))  

> - [2.1 Preview da tabela](#2.1-Preview-da-tabela)  

> - [2.2 Quantidade de linhas no dataset](#2.2-Quantidade-de-linhas-no-dataset)  

> - [2.3 Tipos de dados em cada coluna](#2.3-Tipos-de-dados-em-cada-coluna)  

> - [2.4.1 Categorias da variável *escolaridade*](#2.4.1-Categorias-da-variável-escolaridade)  
> - [2.4.2 Categorias da variável *estado_civil*](#2.4.2-Categorias-da-variável-estado_civil)  
> - [2.4.3 Categorias da variável *sexo*](#2.4.3-Categorias-da-variável-sexo)  

> - [2.5.1 Quantidade de clientes em cada faixa salarial](#2.5.1-Quantidade-de-clientes-em-cada-faixa-salarial)  
> - [2.5.2 Quantidade de clientes de acordo com cada tipo de cartão](#2.5.2-Quantidade-de-clientes-de-acordo-com-cada-tipo-de-cartão)  
> - [2.5.3 Quantidade de clientes de acordo com o sexo](#2.5.3-Quantidade-de-clientes-de-acordo-com-o-sexo)  

> - [2.6.1 Análise da idade dos clientes (média, mínima e máxima)](#2.6.1-Análise-da-idade-dos-clientes-(média,-mínima-e-máxima))  
> - [2.6.2 Análise dos gastos dos clientes (média, mínima e máxima)](#2.6.2-Análise-dos-gastos-dos-clientes-(média,-mínima-e-máxima))  

> - [2.7.1 Limite mínimo de crédito de acordo com o tipo de cartão, sexo e escolaridade](#2.7.1-Limite-mínimo-de-crédito-de-acordo-com-o-tipo-de-cartão,-sexo-e-escolaridade)  
> - [2.7.2 Limite máximo de crédito de acordo com o tipo de cartão, sexo e escolaridade](#2.7.2-Limite-máximo-de-crédito-de-acordo-com-o-tipo-de-cartão,-sexo-e-escolaridade)  

> - [2.8 Análise da quantidade e valores das transações de acordo com o sexo e estado civil](#2.8-Análise-da-quantidade-e-valores-das-transações-de-acordo-com-o-sexo-e-estado-civil)  

> - [2.9 Média da quantidade de transações, valor das transações e limite de crédito de acordo com o tipo de cartão e salário anual](#2.9-Média-da-quantidade-de-transações,-valor-das-transações-e-limite-de-crédito-de-acordo-com-o-tipo-de-cartão-e-salário-anual)  

- [**3. Conclusão**](#3.-Conclusão:)

# **0. Conjunto de dados**

O [conjunto de dados](https://raw.githubusercontent.com/rhatiro/exercicios-SQL-para-Analise-de-Dados-EBAC/main/Mo%CC%81dulo%208%20-%20Projeto%20Final/database/credito8.csv) utilizado neste projeto consiste em uma subamostra do [dataset original](https://raw.githubusercontent.com/andre-marcos-perez/ebac-course-utils/main/dataset/credito.csv) do professor [André Perez](https://www.linkedin.com/in/andremarcosperez/). É importante destacar que pode haver discrepâncias nas proporções das categorias nesta amostra parcial em relação ao conjunto de dados original. Essa subamostra foi extraída com o objetivo de facilitar o processamento dos dados para o projeto em questão. Para acessar o dataset completo, você pode visitar o repositório GitHub do professor [André Perez](https://github.com/andre-marcos-perez), disponível através do seguinte link: https://github.com/andre-marcos-perez/ebac-course-utils/blob/main/dataset/credito.csv

É válido ressaltar que as estimativas baseadas nesse conjunto de dados podem apresentar menor precisão em comparação com a amostra completa. Portanto, os resultados aqui apresentados não possuem a mesma fidelidade de uma análise realizada com o dataset completo. Eles são utilizados apenas para fins didáticos do curso e para demonstração pessoal no portfólio. No entanto, todas as *queries* podem ser igualmente aplicadas ao conjunto de dados original para obter conclusões confiáveis e úteis em análises mais aprofundadas.

Os dados fornecem informações sobre clientes de um banco e incluem as seguintes variáveis:

| Coluna                   | Descrição                                               | Tipo     |
| :----------------------- | :-----------------------------------------------------: | -------: |
| **idade**                | idade do cliente                                        | *int*    |
| **sexo**                 | sexo do cliente (F ou M)                                | *string* |
| **dependentes**          | número de dependentes do cliente                        | *int*    |
| **escolaridade**         | nível de escolaridade do clientes                       | *string* |
| **estado_civil**         | estado civil do cliente                                 | *string* |
| **salario_anual**        | faixa salarial do cliente                               | *string* |
| **tipo_cartao**          | tipo de cartao do cliente                               | *string* |
| **qtd_produtos**         | quantidade de produtos comprados nos últimos 12 meses   | *bigint* |
| **iteracoes_12m**        | quantidade de iterações/transacoes nos ultimos 12 meses | *int*    |
| **meses_inativo_12m**    | quantidade de meses que o cliente ficou inativo         | *int*    |
| **limite_credito**       | limite de credito do cliente                            | *float*  |
| **valor_transacoes_12m** | valor das transações dos ultimos 12 meses               | *float*  |
| **qtd_transacoes_12m**   | quantidade de transacoes dos ultimos 12 meses           | *int*    |

A tabela e as *queries* em linguagem SQL foram criadas através do [Amazon Athena](https://aws.amazon.com/pt/athena/) junto com o [Amazon Simple Storage Service (Amazon S3)](https://aws.amazon.com/pt/s3/).

[**Amazon Web Services**](https://aws.amazon.com/pt/), também conhecido como [**AWS**](https://aws.amazon.com/pt/), é uma plataforma de serviços de computação em nuvem, que formam uma plataforma de computação na nuvem oferecida pela Amazon.com.

[Voltar ao índice](#Índice)

In [ ]:
# Import da biblioteca pandas para visualização dos arquivos CSV resultantes das queries

import pandas as pd


# **1. Criação da tabela**

A primeira etapa é a criação da tabela com o *dataset* armazenado no **Amazon S3 Bucket** por meio de uma *query* no **AWS Athena**, para posteriormente realizar as consultas.

```sql
CREATE EXTERNAL TABLE IF NOT EXISTS default.credito (
	`idade` int,
	`sexo` string,
	`dependentes` int,
	`escolaridade` string,
	`estado_civil` string,
	`salario_anual` string,
	`tipo_cartao` string,
	`qtd_produtos` bigint,
	`iteracoes_12m` int,
	`meses_inativo_12m` int,
	`limite_credito` float,
	`valor_transacoes_12m` float,
	`qtd_transacoes_12m` int
)
ROW FORMAT SERDE 'org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe'
WITH SERDEPROPERTIES (
	'serialization.format' = ',',
	'field.delim' = ','
)
LOCATION 's3://bucket-transacoes-ebac-projetofinal/'
TBLPROPERTIES ('has_encrypted_data' = 'false');
```

[Voltar ao índice](#Índice)

# **2. EDA (Exploratory Data Analysis)**

[Voltar ao índice](#Índice)

## 2.1 Preview da tabela

> **Query 1:**

```sql
SELECT * FROM "default"."credito" limit 10;
```

In [ ]:
df = pd.read_csv('Query_1.csv')
df.head()




idade sexo  dependentes         escolaridade estado_civil   salario_anual  \
0     45    M            3         ensino medio       casado     $60K - $80K   
1     49    F            5             mestrado     solteiro  menos que $40K   
2     51    M            3             mestrado       casado    $80K - $120K   
3     40    F            4         ensino medio           na  menos que $40K   
4     40    M            3  sem educacao formal       casado     $60K - $80K   

  tipo_cartao  qtd_produtos  iteracoes_12m  meses_inativo_12m  limite_credito  \
0        blue             5              3                  1        12691.51   
1        blue             6              2                  1         8256.96   
2        blue             4              0                  1         3418.56   
3        blue             3              1                  4         3313.03   
4        blue             5              0                  1         4716.22   

   valor_transacoes_12m  qtd_transacoes_12m  
0               1144.90                  42  
1               1291.45                  33  
2               1887.72                  20  
3               1171.56                  20  
4                816.08                  28

[Voltar ao índice](#Índice)

## 2.2 Quantidade de linhas no dataset

> **Query 2:**

```sql
SELECT COUNT(*) AS quantidade_linhas
FROM credito;
```

In [ ]:
pd.read_csv('Quantidade.csv')

quantidade_linhas
0               2564

O conjunto de dados completo possui 10.127 linhas, enquanto esta amostra contém 2.564 linhas, representando aproximadamente 25% do dataset original.

[Voltar ao índice](#Índice)

## 2.3 Tipos de dados em cada coluna

> **Query 3:**

```sql
DESCRIBE credito;
```

![query03](https://raw.githubusercontent.com/rhatiro/exercicios-SQL-para-Analise-de-Dados-EBAC/main/Mo%CC%81dulo%208%20-%20Projeto%20Final/img/query03.png)

> Os tipos dos dados no *dataset* podem ser descritos como:
> - `int` representa dados do tipo inteiro (numérico);
> - `string` representa dados do tipo texto ou cadeia de caracteres;
> - `bigint` representa dados do tipo inteiro longo (um número inteiro maior do que o tipo int);
> - `float` representa dados do tipo número de ponto flutuante (um número com parte fracionária).

[Voltar ao índice](#Índice)

## 2.4.1 Categorias da variável `escolaridade`

> **Query 4:**

```sql
SELECT DISTINCT escolaridade
FROM credito;
```

In [ ]:
pd.read_csv('escolaridade.csv')
df

idade sexo  dependentes         escolaridade estado_civil   salario_anual  \
0     45    M            3         ensino medio       casado     $60K - $80K   
1     49    F            5             mestrado     solteiro  menos que $40K   
2     51    M            3             mestrado       casado    $80K - $120K   
3     40    F            4         ensino medio           na  menos que $40K   
4     40    M            3  sem educacao formal       casado     $60K - $80K   
5     44    M            2             mestrado       casado     $40K - $60K   
6     51    M            4                   na       casado         $120K +   
7     32    M            0         ensino medio           na     $60K - $80K   
8     37    M            3  sem educacao formal     solteiro     $60K - $80K   
9     48    M            2             mestrado     solteiro    $80K - $120K   

  tipo_cartao  qtd_produtos  iteracoes_12m  meses_inativo_12m  limite_credito  \
0        blue             5              3                  1        12691.51   
1        blue             6              2                  1         8256.96   
2        blue             4              0                  1         3418.56   
3        blue             3              1                  4         3313.03   
4        blue             5              0                  1         4716.22   
5        blue             3              2                  1         4010.69   
6        gold             6              3                  1        34516.72   
7      silver             2              2                  2        29081.49   
8        blue             5              0                  2        22352.50   
9        blue             6              3                  3        11656.41   

   valor_transacoes_12m  qtd_transacoes_12m  
0               1144.90                  42  
1               1291.45                  33  
2               1887.72                  20  
3               1171.56                  20  
4                816.08                  28  
5               1088.07                  24  
6               1330.87                  31  
7               1538.32                  36  
8               1350.14                  24  
9               1441.73                  32

> Podemos notar que há valores **nulos/ausentes** na variável categórica `escolaridade`, representados pelo valor **`na`**.

[Voltar ao índice](#Índice)

## 2.4.2 Categorias da variável `estado_civil`

> **Query 5:**

```sql
SELECT DISTINCT estado_civil
FROM credito;
```

In [ ]:
pd.read_csv('estado_civil.csv')

estado_civil
0     solteiro
1       casado
2           na
3   divorciado

> Também podemos notar a presença de valores nulos na variável categórica `estado_civil`.

[Voltar ao índice](#Índice)

## 2.4.3 Categorias da variável `sexo`

> **Query 6:**

```sql
SELECT DISTINCT sexo
FROM credito;
```

In [ ]:
pd.read_csv('sexo.csv')

sexo
0    M
1    F

> Diferente das demais colunas, não há valores nulos na variável categórica `sexo`.

[Voltar ao índice](#Índice)

## 2.5.1 Quantidade de clientes em cada faixa salarial

> **Query 7:**

```sql
SELECT salario_anual,
	COUNT(*) AS quantidade
FROM credito
GROUP BY salario_anual
ORDER BY CASE
		salario_anual
		WHEN 'menos que $40K' THEN 1
		WHEN '$40K - $60K' THEN 2
		WHEN '$60K - $80K' THEN 3
		WHEN '$80K - $120K' THEN 4
		WHEN '$120K +' THEN 5 ELSE 6
	END;
```

In [ ]:
pd.read_csv('salario_anual.csv')

salario_anual  quantidade
0  menos que $40K         701
1     $40K - $60K         467
2     $60K - $80K         451
3    $80K - $120K         488
4         $120K +         222
5              na         235

![query07.png](https://raw.githubusercontent.com/rhatiro/exercicios-SQL-para-Analise-de-Dados-EBAC/main/Mo%CC%81dulo%208%20-%20Projeto%20Final/img/query07.png)

> Ao desconsiderar a presença dos 235 valores nulos na variável `salario_anual`, podemos observar que a maioria dos clientes recebem anualmente a quantia abaixo de \\$40.000,00 e a minoria acima de \\$120.000,00.

[Voltar ao índice](#Índice)

## 2.5.2 Quantidade de clientes de acordo com cada tipo de cartão

> **Query 8:**

```sql
SELECT tipo_cartao,
	COUNT(*) AS quantidade
FROM credito
GROUP BY tipo_cartao
ORDER BY quantidade DESC;
```

In [ ]:
pd.read_csv('tipo_cartao.csv')

tipo_cartao  quantidade
0        blue        2453
1      silver          93
2        gold          16
3    platinum           2

![query08.png](https://raw.githubusercontent.com/rhatiro/exercicios-SQL-para-Analise-de-Dados-EBAC/main/Mo%CC%81dulo%208%20-%20Projeto%20Final/img/query08.png)

> A maioria possui o tipo de cartão *blue*, dando a entender que supostamente façam parte dos clientes com faixa de salário anual de menor valor.

[Voltar ao índice](#Índice)

## 2.5.3 Quantidade de clientes de acordo com o sexo

> **Query 9:**

```sql
SELECT sexo,
	COUNT(*) AS quantidade
FROM credito
GROUP BY sexo
ORDER BY quantidade DESC;
```

In [ ]:
pd.read_csv('sexo_quantidade.csv')

sexo  quantidade
0    M        1563
1    F        1001

![query09.png](https://raw.githubusercontent.com/rhatiro/exercicios-SQL-para-Analise-de-Dados-EBAC/main/Mo%CC%81dulo%208%20-%20Projeto%20Final/img/query09.png)

> Neste conjunto de dados, a proporção de clientes do sexo masculino e feminino não está equilibrada. Portanto é algo que deve ser levado em consideração ao relacionar esta variável com as demais.

[Voltar ao índice](#Índice)

## 2.6.1 Análise da idade dos clientes (média, mínima e máxima)

> **Query 10:**

```sql
SELECT ROUND(AVG(idade)) AS idade_media,
	MIN(idade) AS idade_minima,
	MAX(idade) AS idade_maxima
FROM credito;
```

In [ ]:
pd.read_csv('idades.csv')

idade_media  idade_minima  idade_maxima
0         46.0            26            73

## 2.6.2 Análise dos gastos dos clientes (média, mínima e máxima)

> **Query 11:**

```sql
SELECT ROUND(AVG(valor_transacoes_12m), 2) AS valor_medio_transacoes,
	MIN(valor_transacoes_12m) AS valor_minimo_transacoes,
	MAX(valor_transacoes_12m) AS valor_maximo_transacoes
FROM credito;
```

In [ ]:
pd.read_csv('valores.csv')

valor_medio_transacoes  valor_minimo_transacoes  valor_maximo_transacoes
0                 1820.33                   510.16                  4776.58

[Voltar ao índice](#Índice)

## 2.7.1 Limite mínimo de crédito de acordo com o tipo de cartão, sexo e escolaridade

> **Query 12:**

```sql
SELECT COUNT(*) as quantidade,
	MIN(limite_credito) AS limite_credito_maximo,
	tipo_cartao,
	sexo,
	escolaridade
FROM credito
WHERE escolaridade != 'na'
GROUP BY sexo,
	escolaridade,
	tipo_cartao
ORDER BY limite_credito_maximo ASC;
```

In [ ]:
pd.read_csv('limite_credito_maximo.csv')

quantidade  limite_credito_maximo tipo_cartao sexo         escolaridade
0           91                1438.03        blue    F            graduacao
1           92                1438.04        blue    F            doutorado
2          133                1438.07        blue    M            doutorado
3          316                1438.08        blue    F             mestrado
4          148                1438.12        blue    F  sem educacao formal
5          190                1438.17        blue    F         ensino medio
6          203                1438.18        blue    M  sem educacao formal
7          330                1438.26        blue    M         ensino medio
8          158                1438.43        blue    M            graduacao
9          462                1438.45        blue    M             mestrado
10           9               10091.74      silver    F  sem educacao formal
11           6               10110.58      silver    F             mestrado
12           5               10357.97      silver    F            doutorado
13           5               10514.17      silver    F         ensino medio
14           3               10790.91      silver    F            graduacao
15          14               12609.78      silver    M         ensino medio
16          22               15198.71      silver    M             mestrado
17           2               15433.35        gold    F         ensino medio
18           3               15459.46      silver    M            doutorado
19           1               15987.78        gold    F             mestrado
20           7               18431.35      silver    M  sem educacao formal
21           7               26750.60      silver    M            graduacao
22           1               34516.16        gold    M            doutorado
23           1               34516.25    platinum    M             mestrado
24           4               34516.60        gold    M         ensino medio
25           2               34516.91        gold    M             mestrado
26           1               34516.94        gold    M            graduacao
27           1               34516.94        gold    F            graduacao
28           1               34516.96    platinum    F            doutorado

[Voltar ao índice](#Índice)

## 2.7.2 Limite máximo de crédito de acordo com o tipo de cartão, sexo e escolaridade

> **Query 13:**

```sql
SELECT COUNT(*) as quantidade,
	MAX(limite_credito) AS limite_credito_maximo,
	tipo_cartao,
	sexo,
	escolaridade
FROM credito
WHERE escolaridade != 'na'
GROUP BY sexo,
	escolaridade,
	tipo_cartao
ORDER BY limite_credito_maximo DESC;
```

In [ ]:
pd.read_csv('limite_credito_maximo_ desc.csv')

quantidade  limite_credito_maximo tipo_cartao sexo         escolaridade
0          203               34516.99        blue    M  sem educacao formal
1            2               34516.98        gold    M             mestrado
2          462               34516.97        blue    M             mestrado
3            1               34516.96    platinum    F            doutorado
4            7               34516.96      silver    M  sem educacao formal
5            4               34516.95        gold    M         ensino medio
6            1               34516.94        gold    M            graduacao
7            1               34516.94        gold    F            graduacao
8            7               34516.89      silver    M            graduacao
9          330               34516.74        blue    M         ensino medio
10         158               34516.73        blue    M            graduacao
11         133               34516.72        blue    M            doutorado
12          22               34516.66      silver    M             mestrado
13           5               34516.53      silver    F            doutorado
14           3               34516.36      silver    F            graduacao
15          14               34516.34      silver    M         ensino medio
16           3               34516.29      silver    M            doutorado
17           9               34516.26      silver    F  sem educacao formal
18           1               34516.25    platinum    M             mestrado
19           1               34516.16        gold    M            doutorado
20           6               30310.45      silver    F             mestrado
21         190               29543.10        blue    F         ensino medio
22         148               26181.89        blue    F  sem educacao formal
23          92               26053.49        blue    F            doutorado
24         316               24936.22        blue    F             mestrado
25          91               23032.31        blue    F            graduacao
26           1               15987.78        gold    F             mestrado
27           2               15534.85        gold    F         ensino medio
28           5               13882.37      silver    F         ensino medio

[Voltar ao índice](#Índice)

É possível observar que não há uma relação considerável da variável `escolaridade` com o limite de crédito dos clientes. Os resultados dos limites mínimos de acordo com as separações categóricas demonstram que o tipo de cartão *blue* representa a maior parte dos limites mais baixos enquanto os demais representam os mais altos.

A variável `sexo` demonstra que clientes do sexo masculino possuem limites máximos de crédito mais altos enquanto clientes do sexo feminino possuem limites máximos de crédito mais baixos.

## 2.8 Análise da quantidade e valores das transações de acordo com o sexo e estado civil

> **Query 14:**

```sql
SELECT COUNT(*) AS quantidade,
	sexo,
	estado_civil,
	MIN(valor_transacoes_12m) AS valor_minimo_transacoes,
	MAX(valor_transacoes_12m) AS valor_maximo_transacoes,
	AVG(valor_transacoes_12m) AS valor_medio_transacoes
FROM credito
WHERE estado_civil != 'na'
GROUP BY sexo,
	estado_civil
ORDER BY quantidade ASC;
```

In [ ]:
pd.read_csv('quantidade_ ascedente.csv')

quantidade sexo estado_civil  valor_minimo_transacoes  \
0          69    F   divorciado                  1039.44   
1          92    M   divorciado                   781.27   
2         308    F     solteiro                   777.75   
3         458    M     solteiro                   530.36   
4         591    F       casado                   510.16   
5         930    M       casado                   563.93   

   valor_maximo_transacoes  valor_medio_transacoes  
0                  4370.80               2024.6688  
1                  4605.89               1959.7296  
2                  4776.58               2035.2687  
3                  4686.93               1978.7189  
4                  4740.94               1707.7339  
5                  4542.69               1723.0211

> Nesta análise podemos notar que a quantidade de transações tem uma relação considerável com a variável `estado_civil`, sendo que a maior parte foi efetuada por clientes casados e a menor parte por clientes divorciados. Em contrapartida, a média de valor das transações também se diferencia de acordo com o estado civil mas de forma contrária, na qual a média mais alta são dos clientes divorciados e a média mais baixa são dos clientes casados, assim como pode ser constatado na relação dos valores mínimos de transações, dando a entender que a baixa quantidade de transações possívelmente tenha relação com o alto valor das mesmas.
>
> Em relação ao sexo do cliente, a variação não chega a ser tão grande, havendo somente uma leve discrepância entre os valores mínimos das transações, demonstrando ser mais alto entre clientes do sexo feminino que estão divorciadas. Já entre as categorias da variável `estado_civil`, observa-se que clientes solteiros e divorciados do sexo masculino efetuaram uma quantidade maior transações do que clientes do sexo feminino, porém a quantidade é um pouco maior entre clientes casados.

[Voltar ao índice](#Índice)

## 2.9 Média da quantidade de transações, valor das transações e limite de crédito de acordo com o tipo de cartão e salário anual

> **Query 15:**

```sql
SELECT COUNT(*) AS quantidade,
	tipo_cartao,
	salario_anual,
	ROUND(AVG(qtd_transacoes_12m)) AS media_qtd_transacoes,
	ROUND(AVG(valor_transacoes_12m), 2) AS media_valor_transacoes,
	ROUND(AVG(limite_credito), 2) AS media_limite_credito
FROM credito
WHERE salario_anual != 'na'
GROUP BY tipo_cartao,
	salario_anual
ORDER BY CASE
		tipo_cartao
		WHEN 'blue' THEN 1
		WHEN 'silver' THEN 2
		WHEN 'gold' THEN 3 ELSE 4
	END,
	CASE
		salario_anual
		WHEN 'menos que $40K' THEN 1
		WHEN '$40K - $60K' THEN 2
		WHEN '$60K - $80K' THEN 3
		WHEN '$80K - $120K' THEN 4
		WHEN '$120K +' THEN 5 ELSE 6
	END;
```

In [ ]:
pd.read_csv('limite_cred_tipo_cartao.csv')

quantidade tipo_cartao   salario_anual  media_qtd_transacoes  \
0          677        blue  menos que $40K                  43.0   
1          449        blue     $40K - $60K                  42.0   
2          431        blue     $60K - $80K                  41.0   
3          461        blue    $80K - $120K                  40.0   
4          212        blue         $120K +                  40.0   
5           20      silver  menos que $40K                  45.0   
6           17      silver     $40K - $60K                  43.0   
7           15      silver     $60K - $80K                  39.0   
8           22      silver    $80K - $120K                  45.0   
9            9      silver         $120K +                  41.0   
10           4        gold  menos que $40K                  38.0   
11           1        gold     $40K - $60K                  34.0   
12           5        gold     $60K - $80K                  58.0   
13           4        gold    $80K - $120K                  44.0   
14           1        gold         $120K +                  31.0   
15           1    platinum    $80K - $120K                  29.0   

    media_valor_transacoes  media_limite_credito  
0                  1857.24               3797.09  
1                  1841.51               4832.76  
2                  1814.45               8061.38  
3                  1748.57              13736.87  
4                  1707.02              17105.39  
5                  2120.20              12007.81  
6                  1776.94              17870.18  
7                  1610.78              30351.36  
8                  1910.17              34516.40  
9                  1611.85              32341.10  
10                 1502.60              15735.89  
11                 1421.27              23981.32  
12                 2802.68              34516.77  
13                 1823.24              34516.84  
14                 1330.87              34516.72  
15                 1156.61              34516.25

> Nesta análise podemos ver novamente a relação entre o limite de crédito com o tipo de cartão do cliente, sendo que as médias mais baixas de limite são dos clientes que possuem o tipo de cartão *blue*, aumentando de acordo com a faixa salarial anual. As demais médias de quantidade e valor das transações não possuem relação tão considerável com as variáveis deste resultado.

[Voltar ao índice](#Índice)

# **3. Conclusão:**

Essas foram algumas das análises extraídas do conjunto de dados, fornecendo insights sobre o perfil dos clientes e seus comportamentos de transação. Resumidamente, destacam-se os seguintes insights:
- A maioria dos clientes possui um salário anual abaixo de $40.000,00, apresentando o tipo de cartão *blue* e limites de crédito mais baixos. Isso indica que a campanha de marketing direcionada a esse grupo específico pode ter um impacto significativo, considerando sua representatividade na base de clientes.
- Ao analisar a variável sexo, observa-se que os clientes do sexo masculino possuem limites máximos de crédito mais elevados em comparação aos clientes do sexo feminino. No entanto, é importante ressaltar que a proporção de clientes masculinos e femininos na base de dados não está equilibrada, o que pode influenciar essa discrepância.
- Verificou-se que os clientes casados são responsáveis pela maior parte das transações, enquanto os clientes divorciados apresentam uma menor participação. Além disso, os clientes divorciados possuem uma média mais alta de valor das transações, sugerindo uma possível relação entre a menor quantidade de transações e seu valor mais elevado.
- Quando analisados por gênero, os clientes solteiros e divorciados do sexo masculino realizaram mais transações em comparação às clientes do sexo feminino. No entanto, a quantidade de transações entre clientes casados é ligeiramente superior. Esses dados destacam a importância de considerar o estado civil e o gênero ao desenvolver estratégias de marketing personalizadas.
- As médias de limite de crédito mais baixas foram encontradas entre os clientes com o tipo de cartão *blue*. Além disso, em cada categoria de tipo de cartão, a média de limite de crédito aumenta de acordo com a faixa salarial anual do cliente. Essa informação pode ser relevante para ajustar as ofertas e comunicações de acordo com o perfil de cada cliente, considerando sua capacidade de gastos.

Esses insights oferecem uma visão mais completa do comportamento dos clientes e podem servir como base para a tomada de decisões estratégicas, permitindo uma segmentação mais eficiente e personalizada dos esforços de marketing.

**Com base nesta análise geral do conjunto de dados, conclui-se que, para uma possível campanha de marketing direcionada aos clientes mais ativos, é recomendado um foco maior nos clientes casados que possuem o tipo de cartão *blue* e apresentam um salário anual abaixo de $40.000,00. Essa estratégia visa otimizar os esforços de marketing, concentrando-se nos segmentos de clientes com maior potencial de resposta e engajamento, permitindo uma alocação mais eficiente dos recursos disponíveis. Ao direcionar a campanha para essa demografia específica, espera-se aumentar a eficácia das ações de marketing e obter resultados mais favoráveis em termos de aquisição e retenção de clientes.**

[Voltar ao índice](#Índice)

---